pallets/automation-time/src/lib.rs

```diff
@@ -1345,10 +1357,29 @@ pub mod pallet {
 				Self::is_valid_time(*time)?;
 			}
 
+			// Execution fee
 			let fee =
 				Self::calculate_execution_fee(&action, execution_times.len().try_into().unwrap());
-			T::FeeHandler::can_pay_fee(&who, fee.clone())
-				.map_err(|_| Error::<T>::InsufficientBalance)?;
+
+			// XCMP fee
+			let xcmp_fee: u128 = match action {
+				Action::XCMP { para_id, currency_id, encoded_call_weight, .. } =>
+					T::XcmpTransactor::get_xcm_fee(
+						u32::from(para_id),
+						currency_id,
+						encoded_call_weight.clone(),
+					)
+					.unwrap()
+					.saturating_mul(execution_times.len().try_into().unwrap()),
+				_ => 0u32.into(),
+			};
+
+			// Note: will need to account for fees in non-native tokens once we start accepting them
+			T::FeeHandler::can_pay_fee(
+				&who,
+				fee.clone().saturating_add(<BalanceOf<T>>::saturated_from(xcmp_fee.clone())),
```

Comment:
can you rename `fee` to `exeuction_fee` so its more clear

Reply:
done.